dns_pre_lda.scala

```diff
@@ -78,51 +79,97 @@ val l_top_domains = Source.fromFile("top-1m.csv").getLines.map(line => {
+                        dns_qry_type: String,
+                        dns_qry_rcode: String,
+                        dns_sev: Int)
+val scoredFileExists = new java.io.File(feedback_file).exists
+
+var falsepositives : org.apache.spark.rdd.RDD[String]  = if (scoredFileExists){
+    /* dns_scores.csv - feedback file structure
+    
+    0   frame_time             object
+    1   frame_len              object
+    2   ip_dst                 object
+    3   dns_qry_name           object
+    4   dns_qry_class           int64
+    5   dns_qry_type            int64
+    6   dns_qry_rcode           int64
+    7   domain                 object
+    8   subdomain              object
+    9   subdomain_length        int64
+    10  num_periods             int64
+    11  subdomain_entropy     float64
+    12  top_domain              int64
+    13  word                   object
+    14  score                 float64
+    15  query_rep              object
+    16  hh                      int64
+    17  ip_sev                  int64
+    18  dns_sev                 int64
+    19  dns_qry_class_name     object
+    20  dns_qry_type_name      object
+    21  dns_qry_rcode_name     object
+    22  network_context       float64
+    23  unix_tstamp             
+    */    
+    val FrameTimeIndex = 0
+    val UnixTstampIndex = 23
+    val FrameLenIndex = 1
+    val IpDstIndex = 2
+    val DnsQryNameIndex = 3
+    val DnsQryClassIndex = 4
+    val DnsQryTypeIndex = 5
+    val DnsQryRcodeIndex = 6
+    val DnsSevIndex = 18
+    
+    val lines = Source.fromFile(feedback_file).getLines().toArray.drop(1)
+    val feedback : org.apache.spark.rdd.RDD[String] = sc.parallelize(lines)
+    val feedback_df = feedback.map(_.split(",")).map(row => Feedback(row(FrameTimeIndex),
+                                                                               row(UnixTstampIndex),
+                                                                               row(FrameLenIndex).trim.toInt,
+                                                                               row(IpDstIndex),
+                                                                               row(DnsQryNameIndex),
+                                                                               row(DnsQryClassIndex),
+                                                                               row(DnsQryTypeIndex),
+                                                                               row(DnsQryRcodeIndex),
+                                                                               row(DnsSevIndex).trim.toInt )).toDF()
+    val result = feedback_df.filter("dns_sev = 3").select("frame_time","unix_tstamp","frame_len", "ip_dst", "dns_qry_name", "dns_qry_class", "dns_qry_type", "dns_qry_rcode").map(_.mkString(","))
+    val toDuplicate : org.apache.spark.rdd.RDD[String] = result.flatMap(x => List.fill(duplication_factor)(x))
+    toDuplicate
+} else{
+        null
+}
 
 var multidata = {
```

Comment:
there are a lot of var declarations in this code... is this necessary? I bet some can be changed to val declarations

val declarations are simpler to debug and they can sometimes lead to performance increases for distributed code (less locking and synchronization)